pkg/config/configobserver/configobserver.go

```diff
@@ -77,9 +78,15 @@ func (c *Controller) retrieveToken() error {
 			if err := json.Unmarshal(data, &pullSecret); err != nil {
 				klog.Errorf("Unable to unmarshal cluster pull-secret: %v", err)
 			}
-			if auth, ok := pullSecret.Auths["cloud.openshift.com"]; ok && len(auth.Auth) > 0 {
-				klog.V(4).Info("Found cloud.openshift.com token")
-				nextConfig.Token = auth.Auth
+			if auth, ok := pullSecret.Auths["cloud.openshift.com"]; ok {
+				token := strings.TrimSpace(auth.Auth)
+				if strings.Contains(token, "\n") || strings.Contains(token, "\r") {
+					return fmt.Errorf("cluster authorization token is not valid: contains newlines")
+				}
```

Comment:
seeing original error can be useful for debug

Reply:
github on mobile sux, this is for line 115 in insightsclient

Reply:
It's a security leak, since the value of the token is printed to cluster operator and if we gather that then we're leaking their pull secret back to us.